SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/SpatialGDKEditorSchemaGenerator.cpp

```diff
@@ -600,13 +628,19 @@ bool TryLoadExistingSchemaDatabase()
 		if (ActorClassPathToSchema.Num() > 0 && NextAvailableComponentId == SpatialConstants::STARTING_GENERATED_COMPONENT_ID)
 		{
 			UE_LOG(LogSpatialGDKSchemaGenerator, Warning, TEXT("Detected an old schema database, it'll be reset."));
-			ClearGeneratedSchema();
+			ResetSchemaGeneratorState();
```

Comment:
So when running `TryLoadExistingSchemaDatabase`, if we hit this if statement it will start deleting things and resetting things and then return false. That seems quite volotile based on the function name. I think we should change it. 
Also, how come we return false in this case?

Reply:
Yes, that's what I'm addressing now.

Reply:
I've split this function into `ResetSchemaGeneratorStateAndCleanupFolders`, `IsAssetReadOnly` and `LoadGeneratorStateFromSchemaDatabase`.